### PR TITLE
handle some of easy `Expr`s

### PR DESCRIPTION
**Pull request by aviatesk**

### CODE REVIEW

File: src/EscapeAnalysis.jl [hunk 1]
```diff
@@ -250,10 +251,14 @@ function find_escapes(ir::IRCode, nargs::Int)
...
                     end
+                elseif is_meta_expr_head(head)
+                    continue
                 elseif head === :enter || head === :leave || head === :pop_exception
                     continue
+                elseif head === :gc_preserve_begin || head === :gc_preserve_end
+                    continue
                 else # TODO: this is too conservative
-                    add_changes!(stmt.args[2:end], ir, Escape(), changes)
+                    add_changes!(stmt.args, ir, Escape(), changes)
```
Comment (TH3CHARLie) [inline src/EscapeAnalysis.jl:261]: I wonder why this changes? under the current master it will report an error on `ERROR: AssertionError: argextype only works on argument-position values`.  on the case of `@analyze_escapes plot(rand(10, 3)) ` @aviatesk

Comment (aviatesk) [reply src/EscapeAnalysis.jl:261]: We don't handle the expression yet, so escape all the arguments conservatively.

Comment (aviatesk) [reply src/EscapeAnalysis.jl:261]: We need to fix the assertion error separately.

Comment (TH3CHARLie) [reply src/EscapeAnalysis.jl:261]: By adding an argument-position check before the argextpye call?

Comment (aviatesk) [reply src/EscapeAnalysis.jl:261]: Not sure honestly. Please debug where the issue stems from.

Comment (aviatesk) [reply src/EscapeAnalysis.jl:261]: It seems like some `:foreigncall` expression can contain nested `:call` expression, which causes this issue: e.g. `(16, :($(Expr(:foreigncall, :(Core.tuple(:pcre2_get_error_message_8, Base.PCRE.PCRE_LIB)), Int32, svec(Int32, Ptr{UInt8}, UInt64), 0, :(:ccall), Core.Argument(2), :(%15), :(%10), :(%10), :(%1), Core.Argument(2)))))`

**Comment by TH3CHARLie** [comment]
I've added the following code:
```
diff --git a/base/compiler/utilities.jl b/base/compiler/utilities.jl
index ed31e382d1..54937ff70a 100644
--- a/base/compiler/utilities.jl
+++ b/base/compiler/utilities.jl
@@ -3,7 +3,9 @@
 ###########
 # generic #
 ###########
-
+macro show(s)
+    return :(println(stdout, $(QuoteNode(s)), " = ", $(esc(s))))
+end
 if !@isdefined(var"@timeit")
     # This is designed to allow inserting timers when loading a second copy
     # of inference for performing performance experiments.
@@ -227,6 +229,7 @@ function argextype(@nospecialize(x), src, sptypes::Vector{Any}, slottypes::Vecto
         elseif x.head === :copyast
             return argextype(x.args[1], src, sptypes, slottypes)
         end
+        @show x
         @assert false "argextype only works on argument-position values"
     elseif isa(x, SlotNumber)
         return slottypes[(x::SlotNumber).id]
```
and find out the expr that triggers the assertion:
```
:x = Expr(:call, Core.tuple, :(:pcre2_get_error_message_8), Base.PCRE.PCRE_LIB)
```

But I think we handle the `:call` before entering that branch, not sure the reason.

File: src/EscapeAnalysis.jl (hunk 1 above)
Comment (TH3CHARLie) [reply src/EscapeAnalysis.jl:261]: We find the same(?) similar place! Does this foreign call get turned into a call? You can see my log below

Comment (TH3CHARLie) [reply src/EscapeAnalysis.jl:261]: OK, let me try to support `foreigncall`, looks not that hard.
